src/main/java/io/fabric8/maven/docker/access/hc/win/NamedPipe.java

```diff
@@ -0,0 +1,316 @@
+package io.fabric8.maven.docker.access.hc.win;
+
+import java.io.FilterInputStream;
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.RandomAccessFile;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.net.SocketException;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.channels.SocketChannel;
+import java.nio.charset.Charset;
+
+import io.fabric8.maven.docker.util.Logger;
+
+final class NamedPipe extends Socket {
+
+	// Logging
+    private final Logger log;
+	//for development purposes
+	private final static boolean DEBUG = false;
+	
+    private final Object connectLock = new Object();
+    private volatile boolean inputShutdown, outputShutdown;
+
+    private String socketPath;
+    private SocketAddress socketAddress;
+    private RandomAccessFile namedPipe;
+
+    private FileChannel channel;
+        
+    public NamedPipe(Logger log) throws IOException {
+    	this.log = log;
+    }
+
+    @Override
+    public void connect(SocketAddress endpoint) throws IOException {
+        connect(endpoint, 0);
+    }
+
+    @Override
+    public void connect(SocketAddress endpoint, int timeout) throws IOException {
+        if (endpoint == null) {
```

Comment:
Don't think we need a nullpointer check here. First my IDE says it never can be null (don't know its know this  ;-) but more important it will throw a NPE 10 lines later anyway (if so)